libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -166,53 +172,72 @@ const AP_Param::GroupInfo AP_ADSB::var_info[] = {
 AP_ADSB::AP_ADSB()
 {
     AP_Param::setup_object_defaults(this, var_info);
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
     if (_singleton != nullptr) {
         AP_HAL::panic("AP_ADSB must be singleton");
     }
+#endif
     _singleton = this;
+
+#ifdef ADSB_STATIC_CALLSIGN
+    strncpy(&out_state.cfg.callsign, ADSB_STATIC_CALLSIGN, sizeof(out_state.cfg.callsign));
+#endif
 }
 
 /*
  * Initialize variables and allocate memory for array
  */
 void AP_ADSB::init(void)
 {
-    // in_state
-    in_state.vehicle_count = 0;
     if (in_state.vehicle_list == nullptr) {
-        if (in_state.list_size_param != constrain_int16(in_state.list_size_param, 1, ADSB_VEHICLE_LIST_SIZE_MAX)) {
-            in_state.list_size_param.set_and_notify(ADSB_VEHICLE_LIST_SIZE_DEFAULT);
-            in_state.list_size_param.save();
-        }
-        in_state.list_size = in_state.list_size_param;
-        in_state.vehicle_list = new adsb_vehicle_t[in_state.list_size];
+        // sanity check param
+        in_state.list_size_param = constrain_int16(in_state.list_size_param, 1, INT16_MAX);
+
+        in_state.vehicle_list = new adsb_vehicle_t[in_state.list_size_allocated];
 
         if (in_state.vehicle_list == nullptr) {
-            // dynamic RAM allocation of _vehicle_list[] failed, disable gracefully
-            hal.console->printf("Unable to initialize ADS-B vehicle list\n");
-            _enabled.set_and_notify(0);
+            // dynamic RAM allocation of in_state.vehicle_list[] failed
+            _init_failed = true; // this keeps us from constantly trying to init forever in main update
+            gcs().send_text(MAV_SEVERITY_INFO, "ADSB: Unable to initialize ADSB vehicle list");
+            return;
         }
+        in_state.list_size_allocated = in_state.list_size_param;
     }
-
-    furthest_vehicle_distance = 0;
-    furthest_vehicle_index = 0;
-
-    // out_state
-    set_callsign("PING1234", false);
+    in_state.furthest_vehicle_distance = 0;
```

Comment:
Don't need to do these initialisations if init() is only ever going to be called once.